### DISCUSSION TITLE
Pin Mapping on L4R5ZI

### DISCUSSION THREAD

**Discussion by PerennialNovice** [stm32duino/Arduino_Core_STM32#1636] [Q&A] [answered]
I'm still trying to create a custom variant and therefore am looking at the files for Nucelo-L4R5ZI and the corresponding generic variant.
I don't understand how pin mappings are made.

For the generic it seems that ADC1IN_1 is mapped to PA0
https://github.com/stm32duino/Arduino_Core_STM32/blob/b8756be9e028dfa1651591e2d2ca956773189669/variants/STM32L4xx/L4R5Z(G-I)T_L4R7ZIT_L4S5ZIT_L4S7ZIT/variant_generic.h#L18

For nucleo it is mapped to PA3
https://github.com/stm32duino/Arduino_Core_STM32/blob/b8756be9e028dfa1651591e2d2ca956773189669/variants/STM32L4xx/L4R5Z(G-I)T_L4R7ZIT_L4S5ZIT_L4S7ZIT/variant_NUCLEO_L4R5ZI.h#L129

In general the mappings for analog inputs does not correspond to the datasheet where ADC1IN_1 is said to be on pin PC0
![image](https://user-images.githubusercontent.com/85557472/151190062-3a34916b-884b-4433-a37e-abd9f72e4537.png)

Am I making false assumptions or is this an error?
If I got it wrong, please spare a minute to explain.

Thanks!

**Comment by fpistm** [comment, marked as answer]
Hi @MasteringTheMess 
There is no link between `PIN_Ax` and the `ADCx` peripheral and channel.
`PIN_Ax` means the pin is defined to be the analog pin number `Ax` in Arduino naming.

You use it like this for the NUCLEO_L4R5ZI:
`analogRead(A0);` or `analogRead(PA3);` or `analogRead(0);` or `analogRead(D107); `

then the core will search in the `PinMap_ADC` array of the PeripheralPins.c which peripheral have to be used 
https://github.com/stm32duino/Arduino_Core_STM32/blob/b8756be9e028dfa1651591e2d2ca956773189669/variants/STM32L4xx/L4R5Z(G-I)T_L4R7ZIT_L4S5ZIT_L4S7ZIT/PeripheralPins.c#L36-L54

In our case `PA3` is `ADC1_IN8`
https://github.com/stm32duino/Arduino_Core_STM32/blob/b8756be9e028dfa1651591e2d2ca956773189669/variants/STM32L4xx/L4R5Z(G-I)T_L4R7ZIT_L4S5ZIT_L4S7ZIT/PeripheralPins.c#L40

**Reply by PerennialNovice** [reply]
Thanks a lot @fpistm !

Now I understand how this works :)
